src/measure_theory/function/simple_func_dense_lp.lean

```diff
@@ -895,6 +895,60 @@ begin
   exact λ f hf, h_ae hf.coe_fn_to_Lp (Lp.mem_ℒp _) (this (hf.to_Lp f)),
 end
 
+/-- If a set of ae strongly measurable functions is stable under addition and approximates
+characteristic functions in `ℒp`, then it is dense in `ℒp`. -/
+lemma mem_ℒp.induction_dense (hp_ne_top : p ≠ ∞) (h'p : 1 ≤ p) (P : (α → E) → Prop)
+  (h0P : ∀ (c : E) ⦃s : set α⦄, measurable_set s → μ s < ∞ → ∀ {ε : ℝ≥0∞}, ε ≠ 0 →
+    (∃ (g : α → E), snorm (g - s.indicator (λ x, c)) p μ ≤ ε ∧ P g))
+  (h1P : ∀ f g, P f → P g → P (f + g))
+  (h2P : ∀ f, P f → ae_strongly_measurable f μ)
+  {f : α → E} (hf : mem_ℒp f p μ) {ε : ℝ≥0∞} (hε : ε ≠ 0) :
+  ∃ (g : α → E), snorm (f - g) p μ ≤ ε ∧ P g :=
+begin
+  haveI : fact (1 ≤ p) := ⟨h'p⟩,
+  revert f hf ε,
+  refine mem_ℒp.induction hp_ne_top _ _ _ _ _,
+  { assume c s hs hμs ε εpos,
+    rcases h0P c hs hμs εpos with ⟨g, hg, Pg⟩,
```

Comment:
General question: for assigning names to each term in a conjunction, which is the preferred style, `rcases some_term with ⟨p, q, r⟩` or `obtain ⟨p, q, r⟩ := some_term`? (Or even `obtain ⟨p, q, r⟩, from some_term`)? 
You seem to use both; are there specific situations where you would prefer one over the other?

Reply:
`rcases` or `obtain` are essentially equivalent in this case, and there is no preferred style between the two. One big advantage of `obtain` is that you can specify the type to make things more readable. This is what I try to do in long proofs where something interesting is going on. In this particular case, everything is mostly mechanical so I don't think it's worth the added length.